### PR TITLE
unbranchified apollo and user/statistics

### PR DESCRIPTION
**Pull request by giilbert**
# This isnt going to build, every other module in artemis needs to be unbranchified

I tested statistics and links (the link granter's page will crash because its still trying to use branches)

# Changes

1. All `member` -> `user` also changed all the relations
  - eg `MemberComponent` -> `UserComponent`, `/members/:id` -> `/users/:id`
2. All `memberId` -> `userId`
3. Perm `MANAGE_MEMBERS` -> `MANAGE_USERS`
4. API `PATCH /members/metadata` -> `PATCH /users/metadata` (i find every other members/\*\* route not used and they're deleted)
5. Unbranchified a ton of names and queries in apollo

# TODO:
- Branch managers are able to go to `/branch/[slug]/members/[member-id]` and edit other people's statistics -- make it so anywith perms can also do it
- Make roles global
- Clean up the horror in `packages/apollo/src/controllers/users-controller.ts` and the statistics query because i was too lazy to change the Query type!

### CODE REVIEW

File: packages/apollo/src/util/generate-slug.ts [hunk 1]
```diff
@@ -1,9 +1,6 @@
 import { prisma } from "./prisma";
 
-export const generateSlug = async (
-  entity: "branch" | "event",
-  name: string,
-) => {
+export const generateSlug = async (entity: "event", name: string) => {
```
Comment (Dash-L) [inline packages/apollo/src/util/generate-slug.ts:3]: Unless there is something I'm missing we probably don't need `entity` at all anymore but :shrug:

File: packages/artemis/src/modules/users/statistics/value-edit.tsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+    { key: string; userId: string; value: number }
+  >("/users/metadata", "patch", `/users/${query.id}/metadata`);
 
   return (
     <Editable
       defaultValue={md.value.toString()}
       onSubmit={async (value) => {
         await edit({
-          memberId: query.id as string,
+          // TODO: will always fall back to the current user, implement change other's statistics
```
Comment (Dash-L) [inline packages/artemis/src/modules/users/statistics/value-edit.tsx:22]: Is there ever a case where this would need a fallback?

Comment (giilbert) [reply packages/artemis/src/modules/users/statistics/value-edit.tsx:22]: That component will be reused as a way to edit other people's statistics, which needs to be done (it is part of branches routes - which still needs to be deleted).  Right now the user can only edit their own stats

Comment (iDarkLightning) [reply packages/artemis/src/modules/users/statistics/value-edit.tsx:22]: The user shouldnt be able to edit their own stats

File: packages/apollo/src/util/generate-slug.ts (hunk 1 above)
Comment (giilbert) [reply packages/apollo/src/util/generate-slug.ts:3]: I mainly did it for ~~TS to stop complaining~~ future-proofing

**Pull request merged**